src/FileLogger.java

```diff
@@ -0,0 +1,105 @@
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class FileLogger {
+
+    private FileLoggerConfiguration config;
+    private File logFile;
+    private long fileSize;
+
+    public File getLogFile() {
+        return logFile;
+    }
+    public void setLogFile(File logFile) {
+        this.logFile = logFile;
+    }
+
+    public FileLogger(FileLoggerConfiguration config) {
+        this.config = config;
+        this.logFile = new File(config.getFilePath());
+        if (!logFile.exists()) {
+            try {
+                logFile.createNewFile();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public void log(String logLevel, String message) throws FileMaxSizeReachedException {
+        if (logFile.length() + CurrentTime.getCurrentTime().getBytes().length > config.getMaxSizeOfFile()) {
+            String newFileName = "Log_" + CurrentTime.getCurrentTime() + ".txt";
+            logFile = new File(logFile.getParentFile(), newFileName);
+            try {
+                logFile.createNewFile();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            fileSize = 0;
```

Comment:
This variable `fileSize` is redundant, remove it. You can always check the size of a file by invoking `file.length()`